libs/payments/ui/src/lib/client/components/Image/index.tsx

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import NextImage, { getImageProps } from 'next/image';
+import { ComponentProps } from 'react';
+
+// This custom image component is used as a work around to a CSP error
+// due to Next.js's Image component adding an inline style attribute.
+// For more information, and code origin, please refer to the link below
+// https://github.com/vercel/next.js/issues/45184#issuecomment-1988319088
+export function Image(props: ComponentProps<typeof NextImage>) {
```

Comment:
As an alternative to adding this custom component, the CSP config could also be updated in `middleware.ts` to add `unsafe-hashes` to `style-src` and add the relevant hash for this style, which would also resolve the CSP error.

Or this particular CSP error could just be ignored. It doesn't seem to cause any issues.

Reply:
There seem to be 3 approaches to fixing these, and I’d appreciate your input which direction to go with.
1. Don’t fix the CSP errors and document the one’s not being fixed. App functionality doesn’t seem to be impacted.
1. Add an unsafe-hashes and relevant hashes to the style-src CSP rule.
1. For the Next.js image component at least, create a custom wrapper, and remove the inline style. [See comment, and current implementation, here](https://github.com/mozilla/fxa/pull/18920/files#r2109681077) .

Ended up deciding on option 2, unsafe-hashes.